unet-instance-segmentation/experiment/trainer.py

```diff
@@ -87,10 +98,76 @@ def __init__(self):
         self.min_contour_points = 20
 
     def prepare_data_for_training(self):
-        self._download_data()
+        self.has_segmentation_dataset = self.check_mask_or_segmentation_dataset()
```

Comment:
I'd separe this in two function calls
- The first one to a fn named `get_segmentation_dataset` that either returns the dataset or None
- If the fn returns None, then call the second one that get (and returns) the masks and images dataset, and perform the error handling if it can find them
This way it will avoid having to get the datasets multiple times and be a bit more logical

Reply:
Thanks for the suggestions. I updated the code to explicitly get the segmentation and (image-mask) datasets then download them. this way we also only get-dataset once for each of the datasetversions. Please have a look and let me if you have further feedback.